File: Programmers/Lv0/문자열 여러번 뒤집기.py
# 내 풀이
def solution(my_string, queries):
    for x, y in queries:
        count = 0
        sub_string = list(my_string[x:y + 1])
        sub_string.reverse()
        for z in range(x, y + 1):
            my_string = list(my_string)
            my_string[z] = sub_string[count]
            count += 1
            my_string = "".join(my_string)

    return my_string

# 다른 사람 풀이
def solution(my_string, queries):
    answer=list(my_string)
    for s,e in queries:
        answer[s:e+1]=answer[s:e+1][::-1]
    return ''.join(answer)

File: Programmer_Java/Lv0/문자열 여러번 뒤집기.java
// package Programmers;

// public class 문자열 여러번 뒤집기 {
  
// }

// 출처:프로그래머스,
// https://school.programmers.co.kr/learn/courses/30/lessons/181913

// 풀이과정
/*
 * class Solution {
    public String solution(String my_string, int[][] queries) {
        String answer = "";
        String[] string_split = my_string.split("");
        String after_str = "";
        int start;
        for(int[] check_index:queries){
            int s = check_index[0];
            int e = check_index[1];
            after_str ="";
            String now_str = "";
            for(int i=e; i>=s; i--){
                now_str = string_split[i];
                after_str +=now_str;
            }
            
            start = s;
            for(String change_str:after_str.split("")){
                string_split[start] = change_str;
                start++;
            }
            
        
        }
        for(String sum_alpha: string_split){
            answer+=sum_alpha;
        }
        
        return answer;
    }
}
 */

//풀이과정_개선중
/*
 * class Solution {
    public String solution(String my_string, int[][] queries) {
        String answer = "";
        String[] string_split = my_string.split("");
        String after_str = "";
        int start;
        for(int[] check_index:queries){
            int s = check_index[0];
            int e = check_index[1];
            after_str ="";
            String now_str = "";
            for(int i=e; i>=s; i--){
                now_str = string_split[e];
                after_str +=now_str;
            
            start = s;
            for(String change_str:after_str.split("")){
                string_split[start] = change_str;
            }
            }
        
        }
        for(String sum_alpha: string_split){
            answer+=sum_alpha;
        }
        
        return answer;
    }
}
 */